File: FiringRates_python/EventAlignmentTimes.py
# -*- coding: utf-8 -*-


def EventAlignmentTimes(xds, target_dir, target_center, event):

    #%% Run the function according to the event
    
    if 'trial_gocue' in event:
        from GoCueAlignmentTimes import GoCueAlignmentTimes
        Alignment_Times = GoCueAlignmentTimes(xds, target_dir, target_center)
        
    if 'task_onset' in event:
        if xds._lab_data__meta['task_name'] == 'multi_gadget' or xds._lab_data__meta['task_name'] == 'WB':
            from ForceOnsetAlignmentTimes import ForceOnsetAlignmentTimes
            Alignment_Times = ForceOnsetAlignmentTimes(xds, target_dir, target_center)
        else:
            from CursorOnsetAlignmentTimes import CursorOnsetAlignmentTimes
            Alignment_Times = CursorOnsetAlignmentTimes(xds, target_dir, target_center)
    
    if 'trial_end' in event:
        from TrialEndAlignmentTimes import TrialEndAlignmentTimes
        Alignment_Times = TrialEndAlignmentTimes(xds, target_dir, target_center)
        
    if 'cursor_onset' in event:
        from CursorOnsetAlignmentTimes import CursorOnsetAlignmentTimes
        Alignment_Times = CursorOnsetAlignmentTimes(xds, target_dir, target_center)
    
    if 'force_onset' in event:
        from ForceOnsetAlignmentTimes import ForceOnsetAlignmentTimes
        Alignment_Times = ForceOnsetAlignmentTimes(xds, target_dir, target_center)
        
    
    
    
    
    
    return Alignment_Times
    

File: FiringRates_matlab/EventAlignmentTimes.m
function [Alignment_Times] = EventAlignmentTimes(xds, target_dir, target_center, event)

%% Run the function according to the event

if contains(event, 'trial_gocue')
    [Alignment_Times] = GoCueAlignmentTimes(xds, target_dir, target_center);
end

if contains(event, 'task_onset')
    if strcmp(xds.meta.task, 'multi_gadget') || strcmp(xds.meta.task, 'WB')
       [Alignment_Times] = ForceOnsetAlignmentTimes(xds, target_dir, target_center);
    else
        [Alignment_Times] = CursorOnsetAlignmentTimes(xds, target_dir, target_center);
    end
end

if contains(event, 'trial_end')
    [Alignment_Times] = TrialEndAlignmentTimes(xds, target_dir, target_center);
end

if contains(event, 'cursor_onset')
    [Alignment_Times] = CursorOnsetAlignmentTimes(xds, target_dir, target_center);
end

if contains(event, 'cursor_veloc')
    [Alignment_Times] = CursorVelocAlignmentTimes(xds, target_dir, target_center);
end

if contains(event, 'cursor_acc')
    [Alignment_Times] = CursorAccAlignmentTimes(xds, target_dir, target_center);
end

if contains(event, 'force_onset')
    [Alignment_Times] = ForceOnsetAlignmentTimes(xds, target_dir, target_center);
end

if contains(event, 'force_deriv')
    [Alignment_Times] = ForceDerivAlignmentTimes(xds, target_dir, target_center);
end

if contains(event, 'force_max')
    [Alignment_Times] = ForceMaxAlignmentTimes(xds, target_dir, target_center);
end

if contains(event, 'EMG_max')
    [Alignment_Times] = EMGMaxAlignmentTimes(xds, target_dir, target_center);
end

